video/qtvr_decoder.cpp

```diff
@@ -535,12 +564,136 @@ const Graphics::Surface *QuickTimeDecoder::PanoTrackHandler::decodeNextFrame() {
+			break;
+
+		case 4:
+			projectPanorama(kScaleFactorThree, fov, hfov, tiltAngle, panAngle);
+			break;
+
+		case 0:
+			if (_decoder->_isMouseButtonDown || _decoder->_isKeyDown) {
+				projectPanorama(kScaleFactorOne, fov, hfov, tiltAngle, panAngle);
+			} else {
+				projectPanorama(kScaleFactorThree, fov, hfov, tiltAngle, panAngle);
+			}
+			break;
+		
+		default:
+			projectPanorama(kScaleFactorThree, fov, hfov, tiltAngle, panAngle);
+			break;
+		}
+	} else if (_decoder->_transitionMode == kTransitionModeSwing) {
+		swingTransitionHandler();
+	}
 
 	return _projectedPano;
 }
 
+void QuickTimeDecoder::PanoTrackHandler::swingTransitionHandler() {
+	// The number of steps (the speed) of the transition is decided by the _transitionSpeed
+	const int NUM_STEPS = 300 / _decoder->_transitionSpeed;
+
+	// Calculate the step 
+	float stepFOV = (_decoder->_fov - _currentFOV) / NUM_STEPS;
+	float stepHFOV = (_decoder->_hfov - _currentHFOV) / NUM_STEPS;
+
+	float stepTiltAngle = (_decoder->_tiltAngle - _currentTiltAngle) / NUM_STEPS;
+
+	float targetPanAngle = _decoder->_panAngle;
+	float difference = ABS(_currentPanAngle - targetPanAngle);
+	float stepPanAngle;
+
+	// All of this is just because the panorama is supposed to take the smallest path
+	if (targetPanAngle > _currentPanAngle) {
+		if (difference <= 180) {
+			stepPanAngle = difference / NUM_STEPS;
+		} else {
+			stepPanAngle = (- (360 - difference)) / NUM_STEPS;	
+		}
+	} else {
+		if (difference <= 180) {
+			stepPanAngle = (- difference) / NUM_STEPS;
+		} else {
+			stepPanAngle = (360 - difference) / NUM_STEPS;
+		}
+	}
+
+	int16 rectLeft = _decoder->_origin.x;
+	int16 rectRight = _decoder->_origin.y;
+	Common::Point mouse;
+	int mw = _decoder->_width, mh = _decoder->_height;
+
+	for (int i = 0; i < NUM_STEPS; i++) {
+		projectPanorama(kScaleFactorOne, 
+						_currentFOV + i * stepFOV, 
+						_currentHFOV + i * stepHFOV, 
+						_currentTiltAngle + i * stepTiltAngle, 
+						_currentPanAngle + i * stepPanAngle);
+
+		g_system->copyRectToScreen(_projectedPano->getPixels(), 
+									_projectedPano->pitch,
+									rectLeft, 
+									rectRight,
+									_projectedPano->w,
+									_projectedPano->h);
+
+		Common::Event event; 
+		g_system->getEventManager()->pollEvent(event);
+
+		while (g_system->getEventManager()->pollEvent(event)) {
+			if (Common::isMouseEvent(event))
+			mouse = event.mouse;
+
+			// Referenced from video.cpp in testbed engine
+			if (mouse.x >= _decoder->_prevMouse.x && 
+				mouse.x < _decoder->_prevMouse.x + mw &&
+				mouse.y >= _decoder->_prevMouse.y && 
+				mouse.y < _decoder->_prevMouse.y + mh) {
+
+				switch (event.type) { 
+				case Common::EVENT_MOUSEMOVE: 
+					_decoder->handleMouseMove(event.mouse.x - _decoder->_prevMouse.x, event.mouse.y - _decoder->_prevMouse.y);
+					break;
+
+				case Common::EVENT_QUIT:
+					projectPanorama(kScaleFactorThree, _decoder->_fov, _decoder->_hfov, _decoder->_panAngle, _decoder->_panAngle);
```

Comment:
No, you need to quickly quit the execution here, without re-rendering, as this is a request to quit ScummVM completely.